app/grandchallenge/verifications/models.py

```diff
@@ -80,6 +83,29 @@ def save(self, *args, **kwargs):
 
         super().save(*args, **kwargs)
 
+        if self.has_changed("is_verified") and self.is_verified:
+            permission_request_classes = [
+                apps.get_model(
+                    app_label="algorithms",
+                    model_name="AlgorithmPermissionRequest",
+                ),
+                apps.get_model(
+                    app_label="archives", model_name="ArchivePermissionRequest"
+                ),
+                apps.get_model(
+                    app_label="reader_studies",
+                    model_name="ReaderStudyPermissionRequest",
+                ),
+            ]
+
+            for request_class in permission_request_classes:
+                pending_requests = request_class.objects.filter(
+                    user=self.user, status=request_class.PENDING
+                )
```

Comment:
This includes all requests. We only need ones where the `access_request_handling` is `ACCEPT_VERIFIED_USERS`.

Reply:
There is not a test to ensure that only requests with `access_request_handling == ACCEPT_VERIFIED_USERS` are updated.

Reply:
Done

Reply:
btw, process_access_request already checks if the request handling is to accept verified users only ([here](https://github.com/comic/grand-challenge.org/blob/093c5980e57b71f664cdd139997685a84ee07529/app/grandchallenge/core/utils/access_requests.py#L37)), even if we did not filter for them in the verifications model save(). I added the filter anyway to the mentioned code (line 103).

Should I test process_access_request in this case, or the verification model?

Reply:
You should only be modifying requests with `access_request_handling == ACCEPT_VERIFIED_USERS`. You need a filter for that. You need a test for that.

Reply:
Understood. Since the call of this function is within save(), I tested it by making two permission requests (one for object with manual review and the other for accept_verified_users_only). Then checked the status after verification to make sure the request for object with manual review is still pending.